aiomisc/log/formatter/json.py

```diff
@@ -0,0 +1,101 @@
+import json
+import logging
+import sys
+import traceback
+from types import MappingProxyType
+from typing import Union
+
+
+class JSONLogFormatter(logging.Formatter):
+    LEVELS = MappingProxyType({
+        logging.CRITICAL: "crit",
+        logging.FATAL: "fatal",
+        logging.ERROR: "error",
+        logging.WARNING: "warn",
+        logging.WARN: "warn",
+        logging.INFO: "info",
+        logging.DEBUG: "debug",
+        logging.NOTSET: None,
+    })
+
+    FIELD_MAPPING = MappingProxyType({
+        'filename': ('code_file', str),
+        'funcName': ('code_func', str),
+        'lineno': ('code_line', int),
+        'module': ('code_module', str),
+        'name': ('identifier', str),
+        'msg': ('message_raw', str),
+        'process': ('pid', int),
+        'processName': ('process_name', str),
+        'threadName': ('thread_name', str),
+    })
+
+    def __init__(self, fmt=None, datefmt=None, style='%'):
+        super().__init__(
+            datefmt=datefmt if datefmt is not Ellipsis else None,
+            fmt=fmt, style=style
+        )
+
+    def format(self, record: logging.LogRecord):
+        record_dict = MappingProxyType(record.__dict__)
+
+        data = dict(errno=0 if not record.exc_info else 255)
+
+        for key, value in self.FIELD_MAPPING.items():
+            mapping, field_type = value
+
+            v = record_dict.get(key)
+
+            if not isinstance(v, field_type):
+                v = field_type(v)
```

Comment:
Why we need this? It's pretty slow I guess

Reply:
`isinstance` is not so heavy. And the fields mapping has not variable length.